app/controllers/campaign_pages_controller.rb

```diff
@@ -1,8 +1,18 @@
 class CampaignPagesController < ApplicationController
 
+  def index
+    @campaign_pages = CampaignPage.where active: true
+  end
+
   def new
     @campaign_page = CampaignPage.new
     @templates = Template.where active: true
+    @campaigns = Campaign.where active: true
+    # Sets @campaign, which is defined if a new campaign page is created through a link from a campaign page.
+    # In this case, that campaign is set as a default in the dropdown list.
+    @campaign = params[:campaign]
+    # Load the first active template as a default.
+    @template = @templates.first  
```

Comment:
@template = @template.first sets the template to be loaded initially on the page as the first template on the list. I added that, because without it, the first template on the list couldn't be loaded without first selecting another template, and then selecting the first one, because templates get loaded with AJAX requests that get triggered on change of the select menu.